lib/main.dart

```diff
@@ -100,46 +82,56 @@ class _DriblaAppScreenState extends State<DriblaAppScreen> {
             ),
             actions: [
               TextButton(
-                child: const Text("OK"),
                 onPressed: () {
+                  _connect();
                   Navigator.of(context).pop();
                 },
+                child: Text(loc.retryButtonText),
               ),
             ],
           );
         });
   }
 
-  Future<void> _connect(
-      [Duration scanTimeout = const Duration(seconds: 3)]) async {
+  Future<void> _connect({
+    Duration scanTimeout = const Duration(seconds: 30),
+  }) async {
     _flutterBlue.scan(timeout: scanTimeout).listen((scanResult) async {
       if (scanResult.device.name.isNotEmpty && _selectedDevice == null) {
         await scanResult.device.connect();
         List<BluetoothService> bluetoothServices =
             await scanResult.device.discoverServices();
         BluetoothService? service = bluetoothServices
-            .where((element) =>
-                element.uuid.toString() ==
-                'cb421a98-1247-442f-880d-e8259078f1f4')
+            .where(
+              (element) =>
+                  element.uuid.toString() ==
+                  "cb421a98-1247-442f-880d-e8259078f1f4",
```

Comment:
Since the UUIDs are constant, this isn't such a bad thing, but they should perhaps be moved into another class as static constants.